### ISSUE TITLE
UP037: autofix removes quotes from Literal["value"] typing when not imported

### ISSUE THREAD

**Issue by Prometheos2** [astral-sh/ruff#7025]
UP037's autofix removes quotes from typing.Literal["value"] when Literal is not imported, but it's supposed to have str values.

```python
_DTYPE_BACKEND: Literal["pyarrow", "numpy_nullable"]
``` 
turns into 
```python
_DTYPE_BACKEND: Literal[pyarrow, numpy_nullable]
```

## Details
* **Version:** 0.0.286
* **Relevant settings:** `select = ["UP"]`
* **Commands producing the bug:**
	* the VSC extension's autofix
	* `ruff . --fix`
	* `ruff . --fix --isolated --select UP`

**Comment by charliermarsh** [comment]
Is it a mistake or intentional that `Literal` is not imported?

**Comment by zanieb** [comment]
The problem here is that we cannot identify `Literal` as `typing.Literal` if it's not imported. We could add an exception for this to avoid false positives but it could introduce have false negatives. Perhaps we can just avoid applying this fix (or reduce the applicability) if there's not a binding for the parent name of the subscript?

**Comment by charliermarsh** [comment]
Yeah this is largely "working as intended" in that if we can't resolve a symbol to `typing.Literal`, then we treat values within subscripts in type annotations as type annotations themselves (`typing.Literal` is an exception).

I suppose we could change the logic to avoid treating values within subscripts as annotations if the subscript itself is unbound, although it would do the wrong thing if the user has any wildcard imports.

**Comment by zanieb** [comment]
I'm not sure how to implement this with the way that we visit type definitions — do you have a suggestion @charliermarsh ?

**Comment by charliermarsh** [comment]
@zanieb - What if we just treat "Literal" as `typing.Literal` if the symbol is unresolved...?
